wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -94,11 +94,36 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
     private val dashPayProfileDaoAsync = AppDatabase.getAppDatabase().dashPayProfileDaoAsync()
     private val dashPayContactRequestDaoAsync = AppDatabase.getAppDatabase().dashPayContactRequestDaoAsync()
 
+    private val securityGuard = SecurityGuard()
     private lateinit var blockchainIdentity: BlockchainIdentity
+
     private val backgroundThread = HandlerThread("background", Process.THREAD_PRIORITY_BACKGROUND)
-    private val backgroundHandler by lazy {
+    private val backgroundHandler: Handler
+
+    init {
         backgroundThread.start()
-        Handler(backgroundThread.looper)
+        backgroundHandler = Handler(backgroundThread.looper)
+    }
+
+    fun init() {
+        GlobalScope.launch {
+            blockchainIdentityDataDaoAsync.load()?.let {
+                blockchainIdentity = initBlockchainIdentity(it, walletApplication.wallet)
+                while (isActive) {
+                    log.info("Timer: Update contacts")
+                    platformRepoInstance.updateContactRequests()
+                    delay(UPDATE_TIMER_DELAY)
+                }
+            }
```

Comment:
This will load the `blockchainIdentity` and start periodic contacts sync without the need of using Handler.
GlobalScope is active as long as the app is launched.